From d0e84dfad4e749415fc063094626c864052a0a61 Mon Sep 17 00:00:00 2001
From: DuanqiangWen <duanqiangwen@net-swift.com>
Date: Tue, 18 Oct 2022 21:51:36 -0400
Subject: [PATCH] openeuler: configs: delete txgbe/Kconfig, add txgbe_config to
 netswift/Kconfig
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 79611f436ed90e042c3740d5bb0fe0c66829758c
Modified-by-SEL: No

add CONFIG_TXGBE_HWMON for hardware monitoring support
add CONFIG_TXGBE_DEBUG_FS for debug_fs support
add CONFIG_TXGBE_POLL_LINK_STATUS for polling to get phy status support instead of interrupt.
add CONFIG_TXGBE_SYSFS for sysfs support

driver inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5VU60
CVE: NA

Signed-off-by: Duanqiang Wen <duanqiangwen@net-swift.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/netswift/Kconfig       | 54 ++++++++++++++++++++-
 drivers/net/ethernet/netswift/txgbe/Kconfig | 13 -----
 2 files changed, 53 insertions(+), 14 deletions(-)
 delete mode 100644 drivers/net/ethernet/netswift/txgbe/Kconfig

diff --git a/drivers/net/ethernet/netswift/Kconfig b/drivers/net/ethernet/netswift/Kconfig
index 8be1eadc602c..d88312538a4f 100644
--- a/drivers/net/ethernet/netswift/Kconfig
+++ b/drivers/net/ethernet/netswift/Kconfig
@@ -16,6 +16,58 @@ config NET_VENDOR_NETSWIFT
 
 if NET_VENDOR_NETSWIFT
 
-source "drivers/net/ethernet/netswift/txgbe/Kconfig"
+config TXGBE
+	tristate "Netswift PCI-Express 10Gigabit Ethernet support"
+	depends on PCI
+	imply PTP_1588_CLOCK
+	help
+	  This driver supports Netswift 10gigabit ethernet adapters.
+	  For more information on how to identify your adapter, go
+	  to <http://www.net-swift.com>
+
+	  To compile this driver as a module, choose M here. The module
+	  will be called txgbe.
+config TXGBE_HWMON
+	bool "Netswift PCI-Express 10Gigabit adapters HWMON support"
+	default n
+	depends on TXGBE && HWMON && !(TXGBE=y && HWMON=m)
+	help
+	  Say Y if you want to expose thermal sensor data on these devices.
+
+	  If unsure, say N.
+
+config TXGBE_PROCFS
+	bool "Netswift PCI-Express 10Gigabit adapters procfs support"
+	default n
+	depends on TXGBE && !TXGBE_SYSFS
+	help
+	  Say Y if you want to setup procfs for these devices.
+
+	  If unsure, say N.
+
+config TXGBE_DEBUG_FS
+	bool "Netswift PCI-Express 10Gigabit adapters debugfs support"
+	default n
+	depends on TXGBE
+	help
+	  Say Y if you want to setup debugfs for these devices.
+
+	  If unsure, say N.
+	  
+config TXGBE_POLL_LINK_STATUS
+	bool "Netswift PCI-Express 10Gigabit adapters poll mode support"
+	default n
+	depends on TXGBE
+	help
+	  Say Y if you want to turn these devices to poll mode instead of interrupt-trigged TX/RX.
+
+	  If unsure, say N.
+config TXGBE_SYSFS
+	bool "Netswift PCI-Express 10Gigabit adapters sysfs support"
+	default n
+	depends on TXGBE
+	help
+	  Say Y if you want to setup sysfs for these devices.
 
+	  If unsure, say N.
 endif # NET_VENDOR_NETSWIFT
diff --git a/drivers/net/ethernet/netswift/txgbe/Kconfig b/drivers/net/ethernet/netswift/txgbe/Kconfig
deleted file mode 100644
index a735e1be9434..000000000000
--- a/drivers/net/ethernet/netswift/txgbe/Kconfig
+++ /dev/null
@@ -1,13 +0,0 @@
-#
-# Netswift driver configuration
-#
-
-config TXGBE
-	tristate "Netswift 10G Network Interface Card"
-	default n
-	depends on PCI_MSI && NUMA && PCI_IOV && DCB
-	help
-	  This driver supports Netswift 10G Ethernet cards.
-	  To compile this driver as part of the kernel, choose Y here.
-	  If unsure, choose N.
-	  The default is N.
-- 
2.34.1

